### PR TITLE
[EVM] add getBlockByNumber endpoint

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Add endpoint to get block by number (height). Note that the endpoint takes in several special "number" as input:
- earliest -> corresponding to genesis height
- safe/pending/finalized/latest -> all corresponding to latest height since Sei has instant finality

## Testing performed to validate your change
unit tests

### CODE REVIEW

File: evmrpc/block_test.go [hunk 1]
```diff
@@ -109,13 +136,9 @@ func TestGetBlockByHash(t *testing.T) {
 	privKey := hd.Secp256k1.Generate()(derivedPriv)
 	testPrivHex := hex.EncodeToString(privKey.Bytes())
 	key, _ := crypto.HexToECDSA(testPrivHex)
-	evmParams := k.GetParams(ctx)
-	evmParams.ChainConfig.CancunTime = 0 // overwrite to enable cancun
```
Comment (philipsu522) [inline evmrpc/block_test.go:113]: is this not needed anymore?

File: evmrpc/block_test.go [hunk 2]
```diff
@@ -146,3 +164,52 @@ func TestGetBlockByHash(t *testing.T) {
...
+	require.Nil(t, err)
+	b := TxConfig.NewTxBuilder()
+	b.SetMsgs(msg)
+	Tx = b.GetTx()
+	require.Nil(t, EVMKeeper.SetReceipt(Ctx, tx.Hash(), &types.Receipt{
+		From:             "56789",
+		TransactionIndex: 5,
+	}))
+
+	for _, num := range []string{"0x8", "earliest", "latest", "pending", "finalized", "safe"} {
```
Comment (philipsu522) [inline evmrpc/block_test.go:204]: can we test invalid too?

**Review by philipsu522 — APPROVED** [review]

File: evmrpc/block_test.go (hunk 1 above)
Comment (codchen) [reply evmrpc/block_test.go:113]: yeah we don't need this here; it was copied from the ante handler test which tests for different versions

File: evmrpc/block_test.go (hunk 2 above)
Comment (codchen) [reply evmrpc/block_test.go:204]: added

**Pull request merged**